src/V3Broken.cpp

```diff
@@ -295,6 +328,62 @@ class BrokenCheckVisitor final : public AstNVisitor {
         UASSERT_OBJ(
             !(v3Global.assertScoped() && m_inScope && nodep->varp() && !nodep->varScopep()), nodep,
             "VarRef missing VarScope pointer");
+        if (m_cfuncp) {
+            // Check if variable is an in-scope local, otherwise mark as suspect
+            if (const AstVar* const varp = nodep->varp()) {
+                if (!isInScopeLocal(varp)) { m_suspectVarRefs.push_back(nodep); }
+            }
```

Comment:
This will keep pushing the same var onto the list, which might be expensive in space for large statement trees.  Suggest instead m_suspectVarRefs be an unordered set.

Reply:
This pushes the VarRef, and not the Var (so locations are somewhat more helpful), all of which are unique. This does get cleared for each CFunc though, so that bounds the size, so suggest leave this as is.

Reply:
Then use an unordered_map on the var, and store the first VarRef which referenced it?

Reply:
Ok, done that, thanks.